.github/workflows/bench.yml

```diff
@@ -0,0 +1,86 @@
+name: BENCHMARK
+
+# Controls when the action will run.
+on:
+  pull_request:
+    branches: testnet
+
+jobs:
+  hardware-benchmarks:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env
+        run: |
+          echo "HOME=/home/actions" >> $GITHUB_ENV
+          pallet=$(grep "impl pallet_" runtime/src/lib.rs | grep "Config for Runtime" | sed 's/impl pallet_//' | cut -f1 -d: | sort)
+          for i in $pallet; do
+            mkdir -p pallets/$i/src
+          done
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-2022-03-21
+          target: wasm32-unknown-unknown
+          profile: minimal
+          override: true
+
+      - uses: Swatinem/rust-cache@v1
+        continue-on-error: true
+
+      - name: Balances
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: balances
+        run: |
+          mkdir -p pallets/$PALLET/src
+          ./scripts/bench.sh -p $PALLET -cb
```

Comment:
All of these sections can also be implemented as one, and described via a matrix configuration, see https://github.com/gluwa/creditcoin-legacy-consensus-rust/blob/dev/.github/workflows/ci.yml#L47

Reply:
Matrix configuration seems to work at job level only. If it is used, it will create `n` number of jobs, each with its own commit message to the PR.